src/equinixmetalpy/_patch.py

```diff
@@ -0,0 +1,213 @@
+        # type: (PipelineRequest, PipelineResponse) -> None
+        """Logs HTTP response to the DEBUG logger.
+        :param request: The PipelineRequest object.
+        :type request: ~azure.core.pipeline.PipelineRequest
+        :param response: The PipelineResponse object.
+        :type response: ~azure.core.pipeline.PipelineResponse
+        """
+        http_response = response.http_response
+        topdiv = get_top_divider(
+            'Http {} "{}" from {}'.format(
+                http_response.status_code,
+                responses.get(http_response.status_code, "Unknown"),
+                request.http_request.url.replace(self.api_endpoint, ""),
+            )
+        )
+        try:
+            if not _LOGGER.isEnabledFor(logging.DEBUG):
+                return
+
+            _LOGGER.debug(topdiv)
+            log_string = get_headers_string(http_response.headers)
+            # We don't want to log binary data if the response is a file.
+            # log_string += "\nBody:"
+            pattern = re.compile(
+                r'attachment; ?filename=["\w.]+', re.IGNORECASE)
+            header = http_response.headers.get("content-disposition")
+
+            if header and pattern.match(header):
+                filename = header.partition("=")[2]
+                log_string += "\nBody contains file attachments: {}".format(
+                    filename)
+            elif http_response.headers.get("content-type", "").endswith("octet-stream"):
+                log_string += "\nBody contains binary data."
+            elif http_response.headers.get("content-type", "").startswith("image"):
+                log_string += "\nBody contains image data."
+            elif http_response.headers.get("content-type", "").startswith(
+                "application/json"
+            ):
+                log_string += "\nBody:\n" + prettyjson(http_response.text())
+            else:
+                if response.context.options.get("stream", False):
+                    log_string += "\nBody is streamable."
+                else:
+                    txt = http_response.text()
+                    if txt:
+                        log_string += "\nBody:\n{}".format(txt)
+                    else:
+                        log_string += "\nEmpty body"
+                    # log_string += "\n{}".format(http_response.text())
+            _LOGGER.debug(log_string)
+        except Exception as err:  # pylint: disable=broad-except
+            _LOGGER.debug("Failed to log response: %s", repr(err))
+        _LOGGER.debug(get_bottom_divider(topdiv))
+
+
+class Client(GeneratedClient):
+    def __init__(self, credential: str, base_url: str = EM_API, **kwargs: Any) -> None:
+        azure_credential = AzureKeyCredential(credential)
+        if os.getenv("METAL_PYTHON_DEBUG") == "1":
+            kwargs["logging_policy"] = HttpLoggingPolicy(api_endpoint=base_url)
+        sdk_moniker = f"pydo/{_version.VERSION}"
```

Comment:
Anything "pydo" should be Equinix Metal'ified.

Reply:
thanks. I grepped the repo for pydo, it should be fine by now.